lib/std/Target.zig

```diff
@@ -2024,6 +1994,35 @@ pub const Cpu = struct {
     pub fn baseline(arch: Arch, os: Os) Cpu {
         return Model.baseline(arch, os).toCpu(arch);
     }
+
+    /// Returns whether this architecture supports `address_space`. If `context` is `null`, this
+    /// function simply answers the general question of whether the architecture has any concept
+    /// of `address_space`; if non-`null`, the function additionally checks whether
+    /// `address_space` is valid in that context.
+    pub fn supportsAddressSpace(
+        cpu: Cpu,
+        address_space: std.builtin.AddressSpace,
+        context: ?std.builtin.AddressSpace.Context,
+    ) bool {
```

Comment:
A thought I had here: We could pull this all the way up to `std.Target` itself. That way, we could reject e.g. using `ss` when targeting most operating systems on x86 since they don't set it up. @andrewrk do you think this'd be worthwhile, or is that encoding too much knowledge of OS internals into `std.Target` and the compiler?